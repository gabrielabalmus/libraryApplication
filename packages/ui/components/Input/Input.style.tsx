import TextField from "@mui/material/TextField";
import { styled as MuiStyled } from "@mui/material/styles";
import { InputFieldProps } from "./Input.type";

export const StandardInput = MuiStyled(TextField)<InputFieldProps>(
  ({ width, theme }) =>
    theme.unstable_sx({
      maxWidth: { md: width, xs: "100%" },
    })
);
