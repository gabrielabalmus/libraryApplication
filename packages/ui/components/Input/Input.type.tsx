export interface InputProps {
  label: string;
  onChange: (value: string) => void;
  defaultValue?: string;
  type?: InputType;
  errorMessage?: string;
}
export enum InputType {
  text = "text",
  number = "number",
  password = "password",
}
