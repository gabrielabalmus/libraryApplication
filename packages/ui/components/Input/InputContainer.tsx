import { InputProps, InputType } from "./Input.type";
import { StandardInput } from "./Input.style";
import { ChangeEvent } from "react";

const InputContainer: React.FC<InputProps> = ({
  label,
  value,
  type = InputType.text,
  errorMessage,
  onChange,
  width = "inherit",
  multiline = false,
  placeholder = "",
}) => {
  const handleOnChange = (event: ChangeEvent<HTMLInputElement>) => {
    onChange(event.target.value);
  };

  return (
    <StandardInput
      error={!!errorMessage}
      label={label}
      placeholder={placeholder}
      type={type}
      value={value}
      InputProps={{ inputProps: { min: 0 } }}
      variant="standard"
      helperText={errorMessage}
      onChange={handleOnChange}
      width={width}
      multiline={type !==InputType.number && multiline}
      maxRows={4}
    />
  );
};

export default InputContainer;
