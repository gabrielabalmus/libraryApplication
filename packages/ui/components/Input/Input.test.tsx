import { fireEvent, render, screen } from "@testing-library/react";
import Input from ".";
import { InputProps } from "./Input.type";
import { useState } from "react";
import "@testing-library/jest-dom";

const mockChange = jest.fn();

const renderInput = ({ ...props }: InputProps) => render(<Input {...props} />);

describe("snapshot tests for Input", () => {
  it("renders correctly", () => {
    const { container } = renderInput({
      label: "Test1",
      value: "Test input value",
      onChange: mockChange,
      width: "120px",
      multiline: true,
    });

    expect(container).toMatchSnapshot();
  });

  it("renders correctly with error message", () => {
    const { container } = renderInput({
      label: "Test1",
      value: "Test input value",
      onChange: mockChange,
      errorMessage: "Test error message",
    });

    expect(container).toMatchSnapshot();
  });

  it("renders correctly with placeholder", () => {
    const { container } = renderInput({
      label: "Test1",
      value: "Test input value",
      onChange: mockChange,
      placeholder: "Test placeholder",
    });

    expect(container).toMatchSnapshot();
  });
});

const InputState = (other: Omit<InputProps, "onChange">) => {
  const [value, setValue] = useState<string>(other.value);

  return <Input {...other} value={value} onChange={setValue} />;
};

const wrapper = (other: Omit<InputProps, "onChange">) =>
  render(<InputState {...other} />);

describe("logic tests for Input", () => {
  it("takes value correctly", () => {
    wrapper({ value: "", placeholder: "Test placeholder" });

    const input = screen.getByPlaceholderText("Test placeholder");
    expect(input).toBeInTheDocument();
    expect(input).toHaveValue("");

    const value = "Entered Text";
    fireEvent.change(input, {
      target: {
        value,
      },
    });
    expect(input).toHaveValue(value);
  });

  it("display label correctly", () => {
    wrapper({ label: "Test1", value: "" });

    const inputLabel = screen.getByLabelText("Test1");
    expect(inputLabel).toBeInTheDocument();
  });
});
