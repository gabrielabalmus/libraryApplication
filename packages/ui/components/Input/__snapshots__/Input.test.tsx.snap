// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot tests for Input renders correctly 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root css-1m4ah66-MuiFormControl-root-MuiTextField-root"
    width="120px"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for=":r0:"
      id=":r0:-label"
    >
      Test1
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-multiline css-1rb63tl-MuiInputBase-root-MuiInput-root"
    >
      <textarea
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline css-66dh3a-MuiInputBase-input-MuiInput-input"
        id=":r0:"
        min="0"
        placeholder=""
        style="height: 0px; overflow: hidden;"
      >
        Test input value
      </textarea>
      <textarea
        aria-hidden="true"
        class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline css-66dh3a-MuiInputBase-input-MuiInput-input"
        readonly=""
        style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
        tabindex="-1"
      />
    </div>
  </div>
</div>
`;

exports[`snapshot tests for Input renders correctly with error message 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root css-p6zlu0-MuiFormControl-root-MuiTextField-root"
    width="inherit"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-error MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for=":r1:"
      id=":r1:-label"
    >
      Test1
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary Mui-error MuiInputBase-formControl css-v4u5dn-MuiInputBase-root-MuiInput-root"
    >
      <input
        aria-describedby=":r1:-helper-text"
        aria-invalid="true"
        class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        id=":r1:"
        min="0"
        placeholder=""
        type="text"
        value="Test input value"
      />
    </div>
    <p
      class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium MuiFormHelperText-filled css-1d1r5q-MuiFormHelperText-root"
      id=":r1:-helper-text"
    >
      Test error message
    </p>
  </div>
</div>
`;

exports[`snapshot tests for Input renders correctly with placeholder 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root css-p6zlu0-MuiFormControl-root-MuiTextField-root"
    width="inherit"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for=":r2:"
      id=":r2:-label"
    >
      Test1
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-v4u5dn-MuiInputBase-root-MuiInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        id=":r2:"
        min="0"
        placeholder="Test placeholder"
        type="text"
        value="Test input value"
      />
    </div>
  </div>
</div>
`;
